river/sync.go

```diff
@@ -30,6 +31,13 @@ const (
 	fieldTypeDate = "date"
 )
 
+var (
+	expCreateTable = regexp.MustCompile("(?i)^CREATE\\sTABLE(\\sIF\\sNOT\\sEXISTS)?\\s`{0,1}(.*?)`{0,1}\\.{0,1}`{0,1}([^`\\.]+?)`{0,1}\\s.*")
```

Comment:
seem you just copy the logic from go-mysql? Can we export a interface for it now?

Reply:
I think it could be a simple method of Canal after the line "https://github.com/siddontang/go-mysql/blob/master/canal/sync.go#L221".

like this

     var ErrNoQueryTable = errors.New("no query table") // for "CREATE/DROP DATABASE"
     func (c *Canal) GetQueryTable(query []byte) (table []byte, err error)

Reply:
I prefer adding another interface like `OnTableAltered(schema, table String, query String)` instead of https://github.com/siddontang/go-mysql/blob/master/canal/sync.go#L146.

Reply:
any update? @yaoguais 

Can you first send a PR to go-mysql to support it?

Reply:
I think that the method is duplicate with "OnDDL(nextPos mysql.Position, queryEvent *replication.QueryEvent) error".

Reply:
Not duplicated, but enhanced, you may pay attention to it or ignore it.